src/xtgeo/xyz/_polygons_oper.py

```diff
@@ -135,17 +139,17 @@ def _create_boundary_polygon(
             "Your alpha or alpha_factor value is too low, try increasing it!"
         )
 
-    polygons = _get_shapely_polygons_from_edges(coords, edges)
+    pol_coords = _get_polygon_coords_from_edges_shapely(coords, edges)
 
     # sort polygons by their size to get the largest polygons first
-    sorted(polygons, key=lambda pol: len(pol.exterior.coords), reverse=True)
+    pol_coords = sorted(pol_coords, key=len, reverse=True)
 
     data = []
-    for polid, pol in enumerate(polygons):
-        for coord in pol.exterior.coords:
+    for polid, polcoord in enumerate(pol_coords):
+        for coord in polcoord:
             data.append(list(coord) + [polid])
 
-    return data
+    return data or None
```

Comment:
converting an empty list to a None seems superfluous to me.

Reply:
yes maybe. I was just thinking it would then enter this check. But maybe it is better to change that check into  `if not pol` ?https://github.com/equinor/xtgeo/blob/6addf60ff8b63e757bccc9082888f5a5c346bd21/src/xtgeo/surface/_regsurf_boundary.py#L16-L19

Reply:
Yes `not pol` (meaning `[]`) could be simpler here

Reply:
I looked into it a bit more, seems this `if pol is None` test will never be triggered since this method actually returns a xtgeo.Polygons instance and does not evaluate to None. Hence I cannot do the `not pol` either here. 
I suggest moving the check further into the function and remove this obsolete check, agreed @jcrivenaes ?